zerver/forms.py

```diff
@@ -210,8 +211,11 @@ def send_mail(self, subject_template_name, email_template_name,
         the mail through the functions in zerver.lib.send_email, to match
         how we send all other mail in the codebase.
         """
+        request = getattr(self, 'request')
         user = get_user_profile_by_email(to_email)
-        attempted_subdomain = get_subdomain(getattr(self, 'request'))
+        attempted_subdomain = get_subdomain(request)
+        if settings.EMAIL_RATE_LIMITING:
+            rate_limit_email(request, to_email, attempted_subdomain)
```

Comment:
The `domain` argument to these rate_limit functions doesn't have anything to do with subdomains, or DNS -- the name is a little confusing. It's more like "what kind of request is this", which allows different kinds of requests (e.g., expensive and cheap ones) to have independent rate limits.